src/header/common/referer.rs

```diff
@@ -0,0 +1,35 @@
+use header::{Header, HeaderFormat};
+use std::fmt::{self, Show};
+use header::shared::util::from_one_raw_str;
+
+/// The `Referer` header.
+///
+/// The Referer header is used by user agents to inform server about
+/// the page URL user has came from.
+///
+/// See alse [RFC 1945, section 10.13](http://tools.ietf.org/html/rfc1945#section-10.13).
+///
+/// Currently just a string, but maybe better replace it with url::Url or something like it.
```

Comment:
Actually I used `Location` header definition as a model. And yes, I think it would be nice to make it Url, but I'm not sure it will work for every possible value, provided by, e.g., HTTP clients.